service/src/test/java/uk/gov/hmcts/reform/fpl/service/LocalAuthorityUserServiceTest.java

```diff
@@ -28,88 +30,76 @@
 
     private static final String AUTH_TOKEN = "Bearer token";
     private static final String SERVICE_AUTH_TOKEN = "Bearer service token";
-    private static final String JURISDICTION = "PUBLICLAW";
-    private static final String CASE_TYPE = "CARE_SUPERVISION_EPO";
     private static final String CASE_ID = "1";
-    private static final String CREATOR_USER_ID = "1";
+    private static final String[] USER_IDS = {"1", "2", "3"};
     private static final String LOCAL_AUTHORITY = "example";
+    private static final Set<String> caseRoles = Set.of("[LASOLICITOR]","[CREATOR]");
 
     @Mock
     private AuthTokenGenerator authTokenGenerator;
+
     @Mock
-    private CaseAccessApi caseAccessApi;
+    private CaseUserApi caseUserApi;
     @Mock
     private LocalAuthorityUserLookupConfiguration localAuthorityUserLookupConfiguration;
 
+    @Mock
+    private IdamClient client;
+
     @InjectMocks
     private LocalAuthorityUserService localAuthorityUserService;
 
-    @Test
-    void shouldThrowCustomExceptionWhenValidLocalAuthorityHasNoUsers() throws IllegalArgumentException {
-        given(localAuthorityUserLookupConfiguration.getUserIds(LOCAL_AUTHORITY)).willReturn(
-            ImmutableList.<String>builder().build()
-        );
-
-        assertThatThrownBy(() ->
-            localAuthorityUserService.grantUserAccess(AUTH_TOKEN, CREATOR_USER_ID, CASE_ID, LOCAL_AUTHORITY))
-            .isInstanceOf(NoAssociatedUsersException.class)
-            .hasMessage("No users found for the local authority 'example'");
-    }
+    @BeforeEach
+    void setup() {
+        given(client.authenticateUser("fpl-system-update@mailnesia.com", "Password12")).willReturn(AUTH_TOKEN);
 
-    @Test
-    void shouldNotMakeCallToGrantAccessEndpointWhenUserCreatingCaseIsOnlyUserWithinLocalAuthority() {
         given(authTokenGenerator.generate()).willReturn(SERVICE_AUTH_TOKEN);
+
         given(localAuthorityUserLookupConfiguration.getUserIds(LOCAL_AUTHORITY)).willReturn(
             ImmutableList.<String>builder()
-                .add(CREATOR_USER_ID)
+                .add(USER_IDS)
                 .build()
         );
+    }
 
-        localAuthorityUserService.grantUserAccess(AUTH_TOKEN, CREATOR_USER_ID, CASE_ID, LOCAL_AUTHORITY);
+    @Test
+    void shouldMakeCallToUpdateCaseRoleEndpointToGrantAccessRolesToUsersWithinLocalAuthority() {
+        localAuthorityUserService.grantUserAccessWithCaseRole(CASE_ID, LOCAL_AUTHORITY);
 
-        verify(caseAccessApi, never()).grantAccessToCase(
-            any(), any(), any(), any(), any(), any(), any()
-        );
+        verifyUpdateCaseRolesWasCalledForEachUser();
```

Comment:
lets add a number of times to this method, I commented somewhere else something like "...WasCalledOneTimeForEachUser"

Reply:
done :)